crates/iceberg/src/scan.rs

```diff
@@ -503,17 +544,36 @@ pub struct FileScanTask {
     start: u64,
     #[allow(dead_code)]
     length: u64,
+    project_field_id: Vec<usize>,
```

Comment:
Should this be a u64 rather than a usize? Especially if these are intended to be serialized, we'd want to ensure we don't break things when serializing on a 64 bit arch and then deserializing on 32-bit for example.

Probably should call this `projected_field_ids` as well rather than `project_field_id`

Reply:
Good point! I think we can use i32 here as the same of https://github.com/apache/iceberg-rust/blob/ffafe1560b75158fa17ae1c49e5955e4f2f52402/crates/iceberg/src/spec/datatypes.rs#L504